src/main/java/net/earthcomputer/clientcommands/command/MinesweeperCommand.java

```diff
@@ -0,0 +1,385 @@
+                        ticksPlaying = 1;
+                    }
+
+                    click(tileX, tileY);
+
+                    assert minecraft != null && minecraft.player != null;
+                    if (emptyTilesRemaining <= 0) {
+                        minecraft.player.playNotifySound(SoundEvents.NOTE_BLOCK_PLING.value(), SoundSource.MASTER, 1.0f, 2.0f);
+                    } else if (deathCoords != null) {
+                        minecraft.player.playNotifySound(SoundEvents.NOTE_BLOCK_BASS.value(), SoundSource.MASTER, 1.0f, 1.0f);
+                    }
+                } else if (button == InputConstants.MOUSE_BUTTON_RIGHT) {
+                    flag(tileX, tileY);
+                }
+            }
+
+            return true;
+        }
+
+        private boolean gameActive() {
+            return deathCoords == null && emptyTilesRemaining > 0;
+        }
+
+        private void generateMines(int avoidX, int avoidY) {
+            for (int i = 0; i < mines; i++) {
+                int x = random.nextInt(boardWidth);
+                int y = random.nextInt(boardHeight);
+
+                // too close to the clicked position
+                if (Mth.abs(avoidX - x) <= 1 && Mth.abs(avoidY - y) <= 1) {
+                    i--;
+                    continue;
+                }
+
+                // already a mine
+                if ((board[y * boardWidth + x] & 0b1100) >>> 2 == 2) {
+                    i--;
+                    continue;
+                }
+
+                incrementWarning(x - 1, y - 1);
+                incrementWarning(x, y - 1);
+                incrementWarning(x + 1, y - 1);
+
+                incrementWarning(x - 1, y);
+                board[y * boardWidth + x] = 0b000_10_0_0;
+                incrementWarning(x + 1, y);
+
+                incrementWarning(x - 1, y + 1);
+                incrementWarning(x, y + 1);
+                incrementWarning(x + 1, y + 1);
+            }
+        }
+
+        private void incrementWarning(int x, int y) {
+            if (isWithinBounds(x, y)) {
+                int idx = y * boardWidth + x;
+                byte originalTile = board[idx];
+                if ((originalTile & 0b1100) >>> 2 == 1) {
+                    // increment warning quantity
+                    board[idx] += 0b0_001_00_0_0;
+                } else if ((originalTile & 0b1100) >>> 2 == 0) {
+                    // set to warning tile (and since it was empty beforehand, we make it have a quantity of 1)
+                    board[idx] = 0b0_000_01_0_0;
+                }
+            }
+        }
+
+        private boolean isWithinBounds(int x, int y) {
+            return 0 <= x && x < boardWidth && 0 <= y && y < boardHeight;
+        }
+
+        private void click(int x, int y) {
+            byte tile = board[y * boardWidth + x];
+            boolean flagged = (tile & 2) > 0;
+            boolean covered = (tile & 1) == 0;
+            int type = (tile & 0b1100) >>> 2;
```

Comment:
These should still be helper functions, and the type should be made of named constants.